File: 2053-check-if-all-characters-have-equal-number-of-occurrences/2053-check-if-all-characters-have-equal-number-of-occurrences.dart
class Solution {
bool areOccurrencesEqual(String s) {

  Map<String, int> frMap = {};
  
  for (int i = 0; i < s.length; i++) {
    frMap[s[i]] = (frMap[s[i]] ?? 0) + 1;
  }

  final firstCount = frMap.values.first;

  for (int value in frMap.values) {
    if (value != firstCount) {
      return false;
    }
  }

  return true;
}
}

File: 2053-check-if-all-characters-have-equal-number-of-occurrences/2053-check-if-all-characters-have-equal-number-of-occurrences.swift
class Solution {
    func areOccurrencesEqual(_ s: String) -> Bool {
        var freqMap: [Character: Int] = [:]
        
        for char in s {
            freqMap[char, default: 0] += 1
        }
        
        guard let firstCount = freqMap.values.first else {
            return true
        }
        
        
        for value in freqMap.values {
            if value != firstCount {
                return false
            }
        }
        
        return true
    }
}